runelite-client/src/main/java/net/runelite/client/plugins/raids/RaidsOverlay.java

```diff
@@ -176,16 +193,65 @@ else if (plugin.getRoomBlacklist().contains(room.getName().toLowerCase()))
 					}
 
 					name = room == RaidRoom.UNKNOWN_PUZZLE ? "Unknown" : room.getName();
-
+					recommendedItems.add(room.getRecommendedItem());
 					panelComponent.getChildren().add(LineComponent.builder()
-						.left(room.getType().getName())
+						.left(showItems ? "" : room.getType().getName())
 						.right(name)
 						.rightColor(color)
 						.build());
 					break;
 			}
 		}
 
-		return panelComponent.render(graphics);
+		//if size is 1 the only thing in it is -1 which isnt an item
+		if (!showItems || recommendedItems.size() == 1)
```

Comment:
Since `recommendedItems` is used beyond this point to loop over each item ID, it would be better to simply remove `-1` before doing this check. That way, it also does not need to be taken into account later on either.

```suggestion
// Remove -1 which is the item ID placeholder, not a real item
recommendedItems.remove(-1);

if (!showItems || recommendedItems.isEmpty())
```